<a href="https://colab.research.google.com/github/A-R-M-028/IMDB-Dataset-Analysis/blob/main/Project1_SQL_Query_IMDB_Dataset_Analysix.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Dataset
https://drive.google.com/file/d/1xRpPMZ6eD0SdoQlbhGk28k1BsKOLTe7t/view?usp=drive_link

# Importing the libraries

In [89]:
# NumPY -> To perform the Mathematical Operations
import numpy as np

# Pandas -> Data Analysis Tool (DE, DM, DC, DV)
import pandas as pd

# Matplotlib -> Data Visualisations  Tool
import matplotlib.pyplot as plt

# Seaborn -> Data Visualisations  Tool
import seaborn as sns

# ReGex -> Data Cleaning
import re

# SQLite -> Serverless Database
import sqlite3

# Planning/Approach

In [ ]:
# Step1: Connect your database or create a connection -> sqlite3.connect(database_name)
# Step2: Use the cursor function -> database_variable.cursor()

#                                              Workflow

# NB -> Cursors provide a way to navigate through query results, fetching data from Database in a controlled and efficient manner

# 1. You need to establish a connection to the SQLite Database by creating a connecton/connection object
# 2. Then you need to create a cursor object using the cursor() method
# 3. Then execute the query -> cursor_variable.execute('query')
# 4. To fetch the data, then use fetchall() method of the cursor variable/object
# 5. You have to create a DataFrame w.r.t the SQLite
# 6. Data Analysis(EDA, Explatory Data Analysis) -> Data Exploration, Data Manipulation, Data Cleatning, Data Visulaization
# 7. Conclusion at every steps

# Establishing the Connection to the Database

In [90]:
# Storing the path of dataset into a variable 'db
db = ('/content/drive/MyDrive/Projects/movies.sqlite')

# Creting a database connection object
conn = sqlite3.connect(db)

# Creating a cursor object
cur = conn.cursor()

In [91]:
# Fetch all the data of the movies from the movie table in the above database
cur.execute('SELECT * FROM movies')

# Creating a cursor object
movies = cur.fetchall()

In [ ]:
# Displaying all the data from movies table
movies

In [ ]:
# Creating a DataFrame
columns= ['id', 'original_title', 'budget', 'popularity', 'release_data', 'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling', 'uid', 'director_id']
movies_df = pd.DataFrame(movies, columns = columns)

In [ ]:
# Display the dataframe
movies_df

In [ ]:
movies_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4773 entries, 0 to 4772
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              4773 non-null   int64  
 1   original_title  4773 non-null   object 
 2   budget          4773 non-null   int64  
 3   popularity      4773 non-null   int64  
 4   release_data    4773 non-null   object 
 5   revenue         4773 non-null   int64  
 6   title           4773 non-null   object 
 7   vote_average    4773 non-null   float64
 8   vote_count      4773 non-null   int64  
 9   overview        4770 non-null   object 
 10  tagling         3951 non-null   object 
 11  uid             4773 non-null   int64  
 12  director_id     4773 non-null   int64  
dtypes: float64(1), int64(7), object(5)
memory usage: 484.9+ KB


In [ ]:
# Fetch all the data of the directors from the directors table in the above database
cur.execute('SELECT * FROM directors')

# Creating a cursor object
directors = cur.fetchall()

In [ ]:
directors

In [ ]:
# Creating a DataFrame
directors_df = pd.DataFrame(directors, columns = ['name', 'director_id', 'gender', 'uid', 'department'])

In [ ]:
directors_df

name  director_id  gender      uid department
0         James Cameron         4762       2     2710  Directing
1        Gore Verbinski         4763       2     1704  Directing
2            Sam Mendes         4764       2       39  Directing
3     Christopher Nolan         4765       2      525  Directing
4        Andrew Stanton         4766       2        7  Directing
...                 ...          ...     ...      ...        ...
2344      Shane Carruth         7106       2    76624  Directing
2345   Neill Dela Llana         7107       0  1174437  Directing
2346        Scott Smith         7108       0  1219158  Directing
2347        Daniel Hsia         7109       2   208138  Directing
2348   Brian Herzlinger         7110       2    85563  Directing

[2349 rows x 5 columns]

In [ ]:
directors_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2349 entries, 0 to 2348
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   name         2349 non-null   object
 1   director_id  2349 non-null   int64 
 2   gender       2349 non-null   int64 
 3   uid          2349 non-null   int64 
 4   department   2349 non-null   object
dtypes: int64(3), object(2)
memory usage: 91.9+ KB


In [ ]:
movies_df.shape

(4773, 13)

In [ ]:
directors_df.shape

(2349, 5)

In [ ]:
movies_df.describe()

## Q1: How many movies are present in the movies table?

In [ ]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [ ]:
cur.execute('SELECT COUNT(id) FROM movies')
count = cur.fetchall()
print(f'The number of movies availabe in the movies table {count[0][0]}')

The number of movies availabe in the movies table 4773


In [ ]:
# OR -> Error
query = cur.execute('SELECT COUNT(id) FROM movies')
count = cur.fetchall(query)
print(f'The number of movies availabe in the movies table {count[0][0]}')

The number of movies availabe in the movies table 4773


In [ ]:
# Accessing [List[] [tuple] -> Particular value
count[0][0]

4773

## Q2: Find out these three directors ->
1. James Cameron
2. Luc Besson
3. John Woo

In [ ]:
# "SELECT * FROM directors WHERE name = 'James Cameron' OR name = 'Luc Besson' OR name = 'John Woo'"
cur.execute("SELECT * FROM directors WHERE name IN ('James Cameron', 'Luc Besson', 'John Woo')")
three_directors = cur.fetchall()
print(f'Here are the ifno for the asked directors {three_directors}')

Here are the ifno for the asked directors [('James Cameron', 4762, 2, 2710, 'Directing'), ('John Woo', 4893, 2, 11401, 'Directing'), ('Luc Besson', 4949, 2, 59, 'Directing')]


## Q3: Find all the directors with the name starting with 'Steven'

In [ ]:
cur.execute("SELECT * FROM directors WHERE name LIKE 'steven%'")
directors = cur.fetchall()
print(f'Here are the ifno for the asked directors {directors}')

Here are the ifno for the asked directors [('Steven Spielberg', 4799, 2, 488, 'Directing'), ('Steven Soderbergh', 4909, 2, 1884, 'Directing'), ('Steven Brill', 5013, 2, 32593, 'Directing'), ('Steven Zaillian', 5117, 2, 2260, 'Directing'), ('Steven Quale', 5216, 2, 93214, 'Directing'), ('Steven Seagal', 5221, 2, 23880, 'Directing'), ('Steven E. de Souza', 5390, 2, 1726, 'Directing'), ('Steven Shainberg', 5803, 2, 67795, 'Directing'), ('Steven R. Monroe', 6713, 2, 88039, 'Directing')]


## Q4: Count all the female directors
* Hints: gender =1

In [ ]:
directors_df.columns

Index(['name', 'director_id', 'gender', 'uid', 'department'], dtype='object')

In [ ]:
# id OR ANY -> 'name', 'director_id', 'gender', 'uid', 'department'
cur.execute('SELECT COUNT(id) FROM directors WHERE gender = 1')
female_directors = cur.fetchall()
print(f'Here are the ifno for the asked directors {female_directors[0][0]}')

Here are the ifno for the asked directors 150


* Conclusion -> There are 150 female directors in the directors section

## Q5: Find the name of the  10th first female directors

In [ ]:
cur.execute('SELECT name FROM directors where gender = 1')
tenth_female_director = cur.fetchall()
print(f'Here are the all female directors {tenth_female_director}')
# All data given List of tuples [(), () .....]
print(f'Here are the 10th first female directors {tenth_female_director[9][0]}')

Here are the all female directors [('Brenda Chapman',), ('Lilly Wachowski',), ('Jennifer Yuh Nelson',), ('Kathryn Bigelow',), ('Nancy Meyers',), ('Jill Culton',), ('Mimi Leder',), ('Vicky Jenson',), ('Betty Thomas',), ('Angelina Jolie',), ('Karyn Kusama',), ('Phyllida Lloyd',), ('Angela Robinson',), ('Beeban Kidron',), ('Deepa Mehta',), ('Penny Marshall',), ('Julie Taymor',), ('Susan Stroman',), ('Catherine Hardwicke',), ('Julie Anne Robinson',), ('Anne Fletcher',), ('Sam Taylor-Johnson',), ('Joan Chen',), ('Diane Keaton',), ('Sofia Coppola',), ('Elaine May',), ('Dennie Gordon',), ('Jennifer Flackett',), ('Nanette Burstein',), ('Liz Friedlander',), ('Martha Coolidge',), ('Niki Caro',), ('Liliana Cavani',), ('Elizabeth Banks',), ('Callie Khouri',), ('Sanaa Hamri',), ('Jodie Foster',), ('Sharon Maguire',), ('Cathy Malkasian',), ('Kirsten Sheridan',), ('Rachel Talalay',), ('Inna Evlannikova',), ('Bonnie Hunt',), ('Mary McGuckian',), ('Anne Fontaine',), ('Mira Nair',), ('Jessie Nelson',), 

## Q6: Find the top 3 most popular movies

In [ ]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [ ]:
cur.execute('SELECT popularity FROM movies')
top_three_movies= cur.fetchall()
print(f'Top three movies {top_three_movies}')

Top three movies [(150,), (139,), (107,), (112,), (43,), (115,), (48,), (134,), (98,), (155,), (57,), (107,), (145,), (49,), (99,), (53,), (144,), (135,), (52,), (120,), (89,), (37,), (94,), (42,), (61,), (100,), (198,), (64,), (418,), (93,), (35,), (77,), (78,), (3,), (89,), (21,), (116,), (46,), (89,), (73,), (49,), (51,), (59,), (71,), (102,), (81,), (118,), (78,), (43,), (61,), (62,), (56,), (28,), (75,), (51,), (125,), (65,), (66,), (22,), (45,), (39,), (85,), (42,), (67,), (139,), (187,), (92,), (36,), (120,), (32,), (40,), (60,), (90,), (27,), (79,), (44,), (32,), (128,), (94,), (77,), (77,), (110,), (243,), (2,), (41,), (72,), (44,), (130,), (203,), (62,), (47,), (48,), (67,), (69,), (481,), (724,), (167,), (9,), (108,), (6,), (60,), (3,), (127,), (35,), (21,), (56,), (42,), (63,), (202,), (49,), (34,), (25,), (39,), (78,), (101,), (56,), (70,), (53,), (65,), (115,), (44,), (81,), (5,), (73,), (165,), (70,), (99,), (434,), (67,), (86,), (41,), (26,), (44,), (50,), (114,), (21,)

In [ ]:
# Multistring format
cur.execute('''
SELECT original_title
FROM movies
ORDER BY popularity DESC''')
top_three_movies= cur.fetchall()
print(f'Top three movies {top_three_movies[0:3]}')

Top three movies [('Minions',), ('Interstellar',), ('Deadpool',)]


In [ ]:
cur.execute('SELECT original_title FROM movies WHERE popularity IN (875, 724, 514)')
top_three_movies= cur.fetchall()
print(f'Top three movies {top_three_movies[0:3]}')

Top three movies [('Interstellar',), ('Minions',), ('Deadpool',)]


In [ ]:
# Right
cur.execute('SELECT original_title FROM movies ORDER BY popularity DESC LIMIT 3')
top_three_movies= cur.fetchall()
print(f'Top three movies {top_three_movies}')

Top three movies [('Minions',), ('Interstellar',), ('Deadpool',)]


In [ ]:
# To Explore ->
# JOIN
# ORDER BY
# GROUP BY
# SUBQUERY
# AGGREGATION

## Q7: Find the top 3 bankable(budget) movie

In [ ]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [ ]:
cur.execute('SELECT original_title FROM movies ORDER BY budget DESC LIMIT 3')
top_three_movies= cur.fetchall()
print(f'Top three movies {top_three_movies}')

Top three movies [('Pirates of the Caribbean: On Stranger Tides',), ("Pirates of the Caribbean: At World's End",), ('Avengers: Age of Ultron',)]


## Q8: Find out most awarder average rated movie since jan 1st, 2000

In [ ]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [ ]:
# To lookup on release_date
cur.execute('SELECT release_date FROM movies')
rel_date = cur.fetchall()
print(rel_date)

The most awarder average rated movie since jan 1st, 2000 is 2009-12-10


In [ ]:
# To lookup on vote_average
cur.execute('SELECT vote_average from movies')
avg_vote = cur.fetchall()
print(avg_vote)
# Higher to lower -> DESC(top)

[(7.2,), (6.9,), (6.3,), (7.6,), (6.1,), (5.9,), (7.4,), (7.3,), (7.4,), (5.7,), (5.4,), (6.1,), (7.0,), (5.9,), (6.5,), (6.3,), (7.4,), (6.4,), (6.2,), (7.1,), (6.5,), (6.2,), (7.6,), (5.8,), (6.6,), (7.5,), (7.1,), (5.5,), (6.5,), (6.9,), (6.7,), (6.8,), (6.4,), (6.3,), (7.0,), (6.0,), (5.8,), (5.7,), (6.5,), (6.3,), (5.8,), (5.1,), (7.6,), (5.9,), (7.3,), (6.7,), (7.5,), (7.4,), (5.5,), (7.3,), (6.2,), (6.7,), (6.1,), (5.7,), (6.6,), (6.7,), (6.6,), (7.8,), (6.1,), (5.6,), (6.6,), (5.2,), (5.5,), (6.7,), (6.4,), (8.2,), (7.7,), (6.0,), (7.4,), (7.0,), (5.1,), (5.2,), (5.9,), (5.3,), (7.6,), (5.9,), (5.6,), (8.0,), (6.7,), (6.6,), (5.8,), (7.0,), (7.3,), (4.8,), (5.9,), (7.6,), (6.0,), (6.2,), (7.8,), (7.1,), (6.4,), (4.9,), (7.5,), (5.9,), (7.9,), (8.1,), (8.1,), (6.5,), (7.0,), (6.6,), (7.3,), (7.1,), (6.6,), (5.8,), (5.5,), (6.5,), (6.0,), (6.3,), (5.8,), (6.2,), (6.6,), (6.6,), (5.6,), (7.4,), (7.5,), (6.2,), (6.9,), (6.7,), (7.5,), (7.5,), (6.2,), (5.9,), (6.2,), (6.4,), (7.3,),

In [ ]:
# 2 Columns needed -> release_date, vote_average
cur.execute("""
SELECT original_title
FROM movies
WHERE release_date>2000-01-01
ORDER BY vote_average DESC
LIMIT 1
""")
most_awarded = cur.fetchall()
print(f'The most awarder average rated movie since jan 1st, 2000 is {most_awarded[0][0]}')

The most awarder average rated movie since jan 1st, 2000 is Stiff Upper Lips


## Q9: Find all the movies that has been directed by Brenda Chapman

-> KEY?

In [ ]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [ ]:
directors_df.columns

Index(['name', 'director_id', 'gender', 'uid', 'department'], dtype='object')

In [ ]:
cur.execute("""
SELECT original_title
FROM movies
LEFT JOIN directors
ON movies.director_id = directors.id
WHERE directors.name = 'Brenda Chapman'
""")
by_brenda_chapman = cur.fetchall()
print(f'By Brenda Chapman: {by_brenda_chapman}')


By Brenda Chapman: [('Brave',)]


## Q10: Find out the director who has made the most movies

In [ ]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [ ]:
directors_df.columns

Index(['name', 'director_id', 'gender', 'uid', 'department'], dtype='object')

In [ ]:
# JOIN, ORDER BY, GROUP BY
cur.execute("""
SELECT name
FROM directors
JOIN movies
ON directors.id = movies.director_id
GROUP BY director_id
ORDER BY
COUNT(name) DESC
LIMIT 1
""")
mostMovie_director = cur.fetchall()
print(f'Most movie directors: {mostMovie_director}')

Most movie directors: [('Steven Spielberg',)]


## Q11: Find out the directors who is most bankable

In [ ]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [ ]:
directors_df.columns

Index(['name', 'director_id', 'gender', 'uid', 'department'], dtype='object')

In [88]:
# JOIN, ORDER BY, GROUP BY
cur.execute("""
SELECT name
FROM directors
JOIN movies ON directors.id = movies.director_id
GROUP BY directors.id
ORDER BY SUM(movies.budget) DESC
LIMIT 1
""")
most_bankable_director = cur.fetchall()
print(f"Most movie director: {most_bankable_director}")


OperationalError: ignored

## Q12: Find out the top 10 highest budget making movie

In [ ]:
cur.execute("""
SELECT original_title
FROM movies
ORDER BY budget DESC
LIMIT 10
""")
top10 = cur.fetchall()
print(f'Top 10: {top10}')


Top 10: [('Pirates of the Caribbean: On Stranger Tides',), ("Pirates of the Caribbean: At World's End",), ('Avengers: Age of Ultron',), ('Superman Returns',), ('John Carter',), ('Tangled',), ('Spider-Man 3',), ('The Lone Ranger',), ('The Dark Knight Rises',), ('Harry Potter and the Half-Blood Prince',)]


In [ ]:
# COLAB SPACES
!df -h

Filesystem      Size  Used Avail Use% Mounted on
overlay         108G   25G   84G  23% /
tmpfs            64M     0   64M   0% /dev
shm             5.8G     0  5.8G   0% /dev/shm
/dev/root       2.0G 1005M  952M  52% /usr/sbin/docker-init
tmpfs           6.4G   33M  6.4G   1% /var/colab
/dev/sda1        41G   26G   15G  64% /etc/hosts
tmpfs           6.4G     0  6.4G   0% /proc/acpi
tmpfs           6.4G     0  6.4G   0% /proc/scsi
tmpfs           6.4G     0  6.4G   0% /sys/firmware
drive            15G  148M   15G   1% /content/drive


## Q13: Find the top 10 revenue making movies

In [ ]:
cur.execute("""
SELECT original_title
FROM movies
ORDER BY revenue DESC
LIMIT 10""")
top10 = cur.fetchall()
print(f'Top 10 : {top10}')

Top 10 : [('Avatar',), ('Titanic',), ('The Avengers',), ('Jurassic World',), ('Furious 7',), ('Avengers: Age of Ultron',), ('Frozen',), ('Iron Man 3',), ('Minions',), ('Captain America: Civil War',)]


## Q14: Find the top 10 popular movies with highest rating(vote_average)

In [ ]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [ ]:
query = cur.execute("""
SELECT original_title
FROM movies
ORDER BY popularity, vote_average DESC
LIMIT 10
""")
top10 = cur.fetchall()
print(f'Top 10 : {top10}')

Top 10 : [('Stiff Upper Lips',), ('Dancer, Texas Pop. 81',), ('Sardaarji',), ("One Man's Hero",), ('There Goes My Baby',), ('Counting',), ('Déjà Vu',), ('Dolphins and Whales: Tribes of the Ocean',), ('Lake of Fire',), ('Dreaming of Joseph Lees',)]


## Q15: Find out the top 10 directors with the number of movies and revenue where revenue should be taken into account for doing the new analysis

-> O/p: The directors who has got the hisgest revenue should comes at the top and so on and so forth.

* JOIN, ORDER BY

In [ ]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [ ]:
directors_df.columns

Index(['name', 'director_id', 'gender', 'uid', 'department'], dtype='object')

In [95]:
query = cur.execute("""
SELECT name
FROM directors
JOIN movies
ON directors.id = movies.director_id
GROUP BY directors.id
ORDER BY MAX(revenue) DESC
LIMIT 10
""")
top10 = cur.fetchall()
print(f'Top 10 : {top10}')

'''
SELECT directors.name, COUNT(movies.id) AS num_movies, MAX(movies.revenue) AS max_revenue
FROM directors
JOIN movies ON directors.id = movies.director_id
GROUP BY directors.id
ORDER BY MAX(movies.revenue) DESC
LIMIT 10
'''

Top 10 : [('James Cameron',), ('Joss Whedon',), ('Colin Trevorrow',), ('James Wan',), ('Chris Buck',), ('Shane Black',), ('Kyle Balda',), ('Anthony Russo',), ('Michael Bay',), ('Peter Jackson',)]


## Q16: Find out the top 10 directors with the number of movies should be taken into account for doing the new analysis

In [93]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_data',
       'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagling',
       'uid', 'director_id'],
      dtype='object')

In [94]:
directors_df.columns

Index(['name', 'director_id', 'gender', 'uid', 'department'], dtype='object')

In [97]:
cur.execute("""
SELECT directors.name, COUNT(movies.title) AS num_movies
FROM directors
JOIN movies ON directors.id = movies.director_id
GROUP BY directors.id
ORDER BY COUNT(movies.title) DESC
L
""")
top10 = cur.fetchall()
print(f'Top 10 : {top10}')

Top 10 : [('Steven Spielberg', 27), ('Woody Allen', 21), ('Clint Eastwood', 20), ('Martin Scorsese', 20), ('Spike Lee', 16), ('Robert Rodriguez', 16), ('Ridley Scott', 16), ('Renny Harlin', 15), ('Steven Soderbergh', 15), ('Oliver Stone', 14)]


## Q17: Give the title, release date, budget, revenue, popularity, vote average and those movies made by Steven Spielberg

In [103]:
cur.execute("""
SELECT title, release_date, budget, revenue, popularity, vote_average
FROM directors
JOIN movies ON directors.id = movies.director_id
WHERE directors.name = 'Steven Spielberg'
""")
top10 = cur.fetchall()
print(f'Top 10 : {top10}')

Top 10 : [('Indiana Jones and the Kingdom of the Crystal Skull', '2008-05-21', 185000000, 786636033, 75, 5.7), ('The BFG', '2016-06-01', 140000000, 183345589, 44, 6.0), ('War of the Worlds', '2005-06-28', 132000000, 591739379, 48, 6.2), ('The Adventures of Tintin', '2011-10-25', 130000000, 371940071, 89, 6.7), ('Minority Report', '2002-06-20', 102000000, 358372926, 65, 7.1), ('A.I. Artificial Intelligence', '2001-06-29', 100000000, 235926552, 34, 6.8), ('The Lost World: Jurassic Park', '1997-05-23', 73000000, 229074524, 2, 6.2), ('The Terminal', '2004-06-17', 60000000, 219417255, 57, 7.0), ('Munich', '2005-12-22', 70000000, 130358911, 29, 6.9), ('Hook', '1991-12-11', 70000000, 300854823, 33, 6.6), ('War Horse', '2011-12-25', 66000000, 177584879, 29, 7.0), ('Saving Private Ryan', '1998-07-24', 70000000, 481840909, 76, 7.9), ('Lincoln', '2012-11-09', 65000000, 275293450, 36, 6.7), ('Jurassic Park', '1993-06-11', 63000000, 920100000, 40, 7.6), ('Catch Me If You Can', '2002-12-25', 5200000

In [102]:
# OR
cur.execute("""
SELECT movies.title, movies.release_date, movies.budget, movies.revenue, movies.popularity, movies.vote_average, directors.name
FROM directors
JOIN movies ON directors.id = movies.director_id
WHERE directors.name = 'Steven Spielberg'
""")
top10 = cur.fetchall()
print(f'Top 10 : {top10}')

Top 10 : [('Indiana Jones and the Kingdom of the Crystal Skull', '2008-05-21', 185000000, 786636033, 75, 5.7, 'Steven Spielberg'), ('The BFG', '2016-06-01', 140000000, 183345589, 44, 6.0, 'Steven Spielberg'), ('War of the Worlds', '2005-06-28', 132000000, 591739379, 48, 6.2, 'Steven Spielberg'), ('The Adventures of Tintin', '2011-10-25', 130000000, 371940071, 89, 6.7, 'Steven Spielberg'), ('Minority Report', '2002-06-20', 102000000, 358372926, 65, 7.1, 'Steven Spielberg'), ('A.I. Artificial Intelligence', '2001-06-29', 100000000, 235926552, 34, 6.8, 'Steven Spielberg'), ('The Lost World: Jurassic Park', '1997-05-23', 73000000, 229074524, 2, 6.2, 'Steven Spielberg'), ('The Terminal', '2004-06-17', 60000000, 219417255, 57, 7.0, 'Steven Spielberg'), ('Munich', '2005-12-22', 70000000, 130358911, 29, 6.9, 'Steven Spielberg'), ('Hook', '1991-12-11', 70000000, 300854823, 33, 6.6, 'Steven Spielberg'), ('War Horse', '2011-12-25', 66000000, 177584879, 29, 7.0, 'Steven Spielberg'), ('Saving Priva

# Most Important thing to do ->
* JOIN
* ORDER BY, GROUP BY
* SUBQUERY